nilearn/signal.py

```diff
@@ -408,14 +408,9 @@ def clean(signals, detrend=True, standardize=True, confounds=None,
                       (list, tuple, _basestring, np.ndarray, type(None))):
         raise TypeError("confounds keyword has an unhandled type: %s"
                         % confounds.__class__)
-    # Standardize / detrend
-    normalize = False
-    if confounds is not None:
-        # If confounds are to be removed, then force normalization to improve
-        # matrix conditioning.
-        normalize = True
+    # detrend
     signals = _ensure_float(signals)
-    signals = _standardize(signals, normalize=normalize, detrend=detrend)
+    signals = _standardize(signals, normalize=False, detrend=detrend)
```

Comment:
if detrend is False here, the _standardize will just return a copy of the signals without centering them. May be the function _standardize also has to be changed so that the signal is centered even if both standardize and detrend are False ?